internal/service/eks/addon_version_data_source.go

```diff
@@ -0,0 +1,59 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceAddonVersion() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceAddonVersionRead,
+		Schema: map[string]*schema.Schema{
```

Comment:
This schema should include the relevant attributes from [AddonVersionInfo](https://docs.aws.amazon.com/eks/latest/APIReference/API_AddonVersionInfo.html): 

- architecture
- compatibilities (list)

Reply:
This datasource is not returning an `AddonVersionInfo` object but rather a simple string. My understanding of the &schema.Resource.Schema is that it describes the arguments/attributes of the resource/datasource that are allowed to be set or read from.

This datasource allows one to set `addon_name` (string), `kubernetes_version` (string) and `most_recent` (bool). The datasource computes `version` (string) and allows one to read it back. 

Please provide some feedback if I am misunderstanding something.